### PR TITLE
Sidepanel: Keep revisions opening a new instance

### PR DESCRIPTION
**Pull request by gerhardol**
Requires #10145 to be used, but can be reviewed independent of that PR.

## Proposed changes

When launching a new GE instance from the sidepanel, set
first/last selected revisions in the grid as first/second in
the new instance.

### Before

Always selecting HEAD 

image

### After

Keeping selections
image

## Test methodology 

manual

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/BranchTreePanel/SubmoduleNode.cs [hunk 1]
```diff
@@ -86,7 +87,21 @@ public void LaunchGitExtensions()
                 return;
             }
 
-            GitUICommands.LaunchBrowse(workingDir: Info.Path.EnsureTrailingPathSeparator(), ObjectId.WorkTreeId, Info?.Detailed?.RawStatus?.OldCommit);
+            ObjectId? first;
+            ObjectId? second;
+            if (!IsCurrent)
+            {
+                first = ObjectId.WorkTreeId;
+                second = Info?.Detailed?.RawStatus?.OldCommit;
```
Comment (mstv) [inline GitUI/BranchTreePanel/SubmoduleNode.cs:95]: What's the benefit of this old behavior? Why should we have both? Is it clear to the user which one is chosen?
Either remove or add a comment, please.

Comment (gerhardol) [reply GitUI/BranchTreePanel/SubmoduleNode.cs:95]: Old behavior?
This adds the selection when current module, the other is when a submodule is clicked, then select the commited commit and the worktree, i.e. the "diff" for the submodule.

Comment (mstv) [reply GitUI/BranchTreePanel/SubmoduleNode.cs:95]: Thank you for the clarification. I think future readers would appreciate a comment like `open a "diff" of submodule changes`, too.

File: GitUI/GitUICommands.cs [hunk 2]
```diff
@@ -1905,6 +1905,11 @@ public void BrowseSetWorkingDir(string path, ObjectId? selectedId = null, Object
             BrowseRepo?.SetWorkingDir(path, selectedId, firstId);
         }
 
+        public IReadOnlyList<GitRevision> GetSelectedRevisions()
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply GitUI/GitUICommands.cs:1908]: RevisionGrid.GetSelectedRevisions() should never be null, not changing

File: GitUI/BranchTreePanel/SubmoduleNode.cs [hunk 3]
```diff
@@ -86,7 +87,21 @@ public void LaunchGitExtensions()
                 return;
             }
 
-            GitUICommands.LaunchBrowse(workingDir: Info.Path.EnsureTrailingPathSeparator(), ObjectId.WorkTreeId, Info?.Detailed?.RawStatus?.OldCommit);
+            ObjectId? first;
+            ObjectId? second;
+            if (!IsCurrent)
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply GitUI/BranchTreePanel/SubmoduleNode.cs:92]: I kept the existing behavior first, but changing
Also mixed up first/second

File: GitUI/BranchTreePanel/SubmoduleNode.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/BranchTreePanel/SubmoduleNode.cs:95]: added comment

File: GitUI/GitUICommands.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (mstv) [reply GitUI/GitUICommands.cs:1908]: But this function returns `BrowseRepo?.GetSelectedRevisions()`. 🤔

[the remark this answers is not in the document]
Comment (gerhardol) [reply GitUI/GitUICommands.cs]: OK, changed. No other users

**Review by mstv — APPROVED** [review]
LGTM

**Pull request merged**